src/simple/multiplex/client.rs

```diff
@@ -28,20 +28,30 @@ pub trait ClientProto<T: 'static>: 'static {
     /// Response messages.
     type Response: 'static;
 
+    /// The type of request ids to used to correlate requests to responses
+    type RequestId: RequestId;
```

Comment:
With the the change you suggested and assuming [I understood it correctly](https://github.com/tokio-rs/tokio-proto/pull/148#discussion_r105286531) the changes for `u64` tokio-line should now be just a single added line, unless `RequestId` alias was used explicitly like in the examples.

I just pushed [a branch with necessary changes](https://github.com/tokio-rs/tokio-line/compare/master...koivunej:track-generic-requestid) to check that it so it really is just the assoc type.